hadoop-ozone/dist/src/main/smoketest/recon/recon-api.robot

```diff
@@ -57,6 +59,21 @@ Check http return code
                             Should contain      ${result}       200
                         END
 
+Check if the listKeys api responds OK
+    [Arguments]     ${volume}    ${bucket}
+    ${result} =     Curl    "${API_ENDPOINT_URL}/keys/listKeys?startPrefix=/${volume}/${bucket}&limit=1000"
+    Should contain  ${result}   "OK"
+    Should contain  ${result}   "keys"
+    Should contain  ${result}   "${volume}"
+    Should contain  ${result}   "${bucket}"
+
+Curl
+    [Arguments]     ${url}
+    ${curl_command} =    Set Variable    curl -LSs ${url}
+    Run Keyword if      '${SECURITY_ENABLED}' == 'true'     Set Variable    ${curl_command}    curl --negotiate -u : -LSs ${url}
```

Comment:
[`Set Variable`](https://robotframework.org/robotframework/6.1.1/libraries/BuiltIn.html#Set%20Variable) returns the value, does not modify its first argument.

This can be fixed by using [`Set Variable If`](https://robotframework.org/robotframework/6.1.1/libraries/BuiltIn.html#Set%20Variable%20If):

```suggestion
    ${curl_command} =    Set Variable If    '${SECURITY_ENABLED}' == 'true'    curl --negotiate -u : -LSs ${url}    curl -LSs ${url}
```

But other test cases use `curl --negotiate -u :` unconditionally, so maybe we don't need the conditional logic at all.

Reply:
I added code to kinit and to use curl directly without any conditions. It should hopefully pass now; I checked out a separate branch with the same code and ran CI successfully on my fork (just for testing): https://github.com/siddhantsangwan/ozone/actions/runs/12157989638